models/DataObject/ClassDefinition/CustomLayout/Listing/Dao.php

```diff
@@ -53,9 +51,41 @@ public function load()
     public function getTotalCount()
     {
         try {
-            return (int) $this->db->fetchOne('SELECT COUNT(*) FROM custom_layouts ' . $this->getCondition(), $this->model->getConditionVariables());
+            return count($this->loadIdList());
         } catch (\Exception $e) {
             return 0;
         }
     }
+
+    /**
+     * @TODO remove in Pimcore 11
+     *
+     * @return array
+     */
+    protected function loadIdList(): array
+    {
+        $list = parent::loadIdList();
+
+        return array_merge($list, array_keys($this->loadLegacyConfigs()));
+    }
+
+    /**
+     * @TODO remove in Pimcore 11
+     *
+     * @return array
+     */
+    private function loadLegacyConfigs(): array
```

Comment:
```suggestion
    private function loadLegacyConfigs(): array|null
```
in the case that there are no existing custom layouts, it returns `null`

Reply:
I think the $layout value should be changed to `$layouts = [];` instead, as this is passed to array_keys..